crates/relayer-framework/src/tests/relayer_mock/base/types/runtime.rs

```diff
@@ -0,0 +1,51 @@
+use alloc::boxed::Box;
+use async_trait::async_trait;
+use core::{future::Future, time::Duration};
+use std::time::Instant;
+use tokio::time::sleep;
+
+use ibc_relayer_runtime::tokio::context::TokioRuntimeContext;
+
+use crate::{
+    base::one_for_all::traits::runtime::{LogLevel, OfaRuntime},
+    tests::relayer_mock::base::error::Error,
+};
+
+pub type MockRuntimeContext = TokioRuntimeContext<Error>;
+
+#[async_trait]
+impl OfaRuntime for MockRuntimeContext {
+    type Error = Error;
+
+    type Time = Instant;
+
+    async fn log(&self, level: LogLevel, message: &str) {
+        match level {
+            LogLevel::Error => tracing::error!(message),
+            LogLevel::Warn => tracing::warn!(message),
+            LogLevel::Info => tracing::info!(message),
+            LogLevel::Debug => tracing::debug!(message),
+            LogLevel::Trace => tracing::trace!(message),
+        }
+    }
+
+    async fn sleep(&self, duration: Duration) {
+        sleep(duration).await;
```

Comment:
Sleeping is a little tricky to handle, as they makes the test non-deterministic. We might need a sleep queue so that we can manually wake the tasks that are sleeping. Something like

```rust
struct MockRuntime {
  current_time: Timestamp,
  sleep_tasks: Arc<Mutex<Vec<(Timestamp, SenderOnce<()>)>>>,
}

async fn sleep(&self, duration: Duration) {
  let wake_time = self.current_time.add(duration);
  let (sender, receiver) = channel_once();
  self.sleep_tasks.lock().push((wake_time, sender));
  receiver.await;
}
```